utils/update-next.js

```diff
@@ -1,55 +1,133 @@
+/* eslint-disable no-console */
+
+const { dirname, join } = require('path');
 const { execFileSync } = require('child_process');
+const { glob } = require('glob');
+const { pathExists, readJson, writeJson } = require('fs-extra');
 
 function exec(cmd, args, opts) {
-  // eslint-disable-next-line no-console
-  console.log({ input: `${cmd} ${args.join(' ')}` });
+  console.log({ input: `${cmd} ${args.join(' ')}`, cwd: opts?.cwd });
   const output = execFileSync(cmd, args, opts).toString().trim();
   console.log({ output });
   console.log();
   return output;
 }
 
-module.exports = async ({ github, context }) => {
-  const oldVersion = require('../examples/nextjs/package.json').dependencies
-    .next;
+module.exports = async ({ github, context } = {}) => {
   const newVersion = exec('npm', ['view', 'next', 'dist-tags.latest']);
   const branch = `next-${newVersion.replaceAll('.', '-')}`;
 
-  if (oldVersion === newVersion) {
-    // eslint-disable-next-line no-console
+  if (
+    github &&
+    exec('git', ['ls-remote', '--heads', 'origin', branch]).toString().trim()
+  ) {
+    console.log(`Branch ${branch} already exists, skipping update.`);
+    return;
+  }
+
+  let updatedCount = 0;
+
+  // update `examples/nextjs`
+  const oldVersion = require('../examples/nextjs/package.json').dependencies
+    .next;
+  if (github && oldVersion !== newVersion) {
+    updatedCount++;
+    exec('rm', ['-rf', './examples/nextjs']);
+    exec('npx', ['--yes', 'create-next-app@latest', './examples/nextjs']);
+    exec('git', [
+      'config',
+      '--global',
+      'user.email',
+      'infra+release@vercel.com',
+    ]);
+    exec('git', ['config', '--global', 'user.name', 'vercel-release-bot']);
+    exec('git', ['checkout', 'main']);
+    exec('git', ['checkout', '-b', branch]);
+    exec('git', ['add', '-A']);
+    exec('git', ['commit', '-m', branch]);
+    exec('git', ['push', 'origin', branch]);
+  }
+
+  // update tests
+  const testGlobs = [
+    'packages/cli/test/dev/fixtures/*/package.json',
+    'packages/cli/test/fixtures/unit/**/package.json',
+    'packages/fs-detectors/test/fixtures/**/package.json',
+    'packages/next/test/fixtures/*/package.json',
+    'packages/next/test/integration/*/package.json',
+    'packages/static-build/test/fixtures/*/package.json',
+  ];
+  for (const pattern of testGlobs) {
+    const files = await glob(pattern);
+    for (const file of files) {
+      const pkgJson = await readJson(file);
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** add ‏guard

I get an error here when I try this. It looks like the globs go too deep if there are `node_modules` in a fixture. I added this, then it worked:

```
if (file.includes('node_modules')) {
  // don't go into node_modules of fixtures
  return;
}
```

Reply:
Good catch! I've been continually resetting my repo when testing this PR!

Reply:
Fixed!